include/deal.II/base/parameter_handler.h

```diff
@@ -1628,18 +1628,33 @@ class ParameterHandler : public Subscriptor
    * will stop parsing lines after encountering @p last_line .
    * This is handy when adding extra data that shall be parsed manually.
    *
-   * Return whether the read was successful.
+   * @deprecated This function has been deprecated in favor of the replacement
+   * ParameterHandler::parse_input, which raises exceptions to indicate errors
+   * instead of returning an error code.
    */
   virtual bool read_input (std::istream &input,
                            const std::string &filename = "input file",
-                           const std::string &last_line = "");
+                           const std::string &last_line = "") DEAL_II_DEPRECATED;
+
+  /**
+   * Parse each line from a stream until the stream returns the <tt>eof</tt>
+   * condition or error to provide values for known parameter fields. The second
+   * argument can be used to denote the name of the file (if that's what the
+   * input stream represents) we are reading from; this is only used when
+   * creating output for exceptions.
+   *
+   * If non-empty @p last_line is provided, the ParameterHandler object
+   * will stop parsing lines after encountering @p last_line .
+   * This is handy when adding extra data that shall be parsed manually.
+   */
+  virtual void parse_input (std::istream &input,
+                            const std::string &filename = "input file",
+                            const std::string &last_line = "");
```

Comment:
`last_line` was recently added by @davydden I believe and it does something useful.

I don't have strong feelings about the name but I think `parse_input` is slightly (very slightly) more descriptive since the name implies that it is parsing some sort of input file or input stream.